src/zinit/mod.rs

```diff
@@ -0,0 +1,311 @@
+use serde::{de::DeserializeOwned, Deserialize};
+use serde_yaml::{self, Value};
+use std::collections::HashMap;
+use thiserror::Error;
+use tokio::io::{AsyncReadExt, AsyncWriteExt};
+use tokio::net::UnixStream;
+use tokio::time;
+pub mod signals;
+use signals::Signals;
+use std::path::Path;
+
+pub struct Client {
+    socket: String,
+}
+
+const DEFAULT_SOCKET_PATH: &'static str = "/var/run/zinit.sock";
+const DEFAULT_ZINIT_PATH: &'static str = "/etc/zinit";
+
+#[derive(Deserialize)]
+struct CommandResult {
+    state: State,
+    body: Value,
+}
+
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum State {
+    Ok,
+    Error,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceStatus {
+    pub name: String,
+    pub pid: u64,
+    pub state: ServiceState,
+    pub target: ServiceTarget,
+    pub after: Vec<String>,
+}
+
+#[derive(Deserialize)]
+pub struct ServiceState {
+    pub state: PossibleState,
+    pub reason: String,
+}
+
+#[derive(Deserialize, PartialEq)]
+pub enum PossibleState {
```

Comment:
Did you actually test if this will be able to parse the output from the zinit command correctly? I doubt they will. 
Please add tests with real output data from zinit to test this